### PR TITLE
Cluster member type unification

### PR DESCRIPTION
**Pull request by alekece**
### Description
This PR aims to unify the node type representation over all the codebase:
- Turn `Member.type`  string to a protobuffer enumeration
- Turn `ClusterMember.node_type` string to a python enumeration

### How was this PR tested?
I added new tests to test theses changes, plus ran locally the cluster manager and an ingest node and check the communication between the two.

### CODE REVIEW

File: nucliadb/nucliadb/ingest/orm/node.py [hunk 1]
```diff
@@ -54,11 +55,60 @@
...
+        elif label in "Ingest":
+            return NodeType.INGEST
+        elif label in "Train":
+            return NodeType.TRAIN
+        else:
+            logger.warn(f"Unknown '{label}' node type")
+            return NodeType.UNKNOWN
+
+    @staticmethod
+    def from_compat(node_type):
```
Comment (lferran) [inline nucliadb/nucliadb/ingest/orm/node.py:80]: would be nice to add type hints here, so it's clearer from what we are converting to

Comment (lferran) [reply nucliadb/nucliadb/ingest/orm/node.py:80]: or name it `from_pb`?

File: nucliadb/nucliadb/ingest/orm/node.py [hunk 2]
```diff
@@ -54,11 +55,60 @@
...
+        elif node_type == Member.Type.INGEST:
+            return NodeType.INGEST
+        elif node_type == Member.Type.TRAIN:
+            return NodeType.TRAIN
+        elif node_type == Member.Type.UNKNOWN:
+            return NodeType.UNKNOWN
+        else:
+            raise ValueError(f"incompatible node type '{node_type}'")
+
+    def compat(self):
```
[the remark this answers is not in the document]
Comment (alekece) [reply nucliadb/nucliadb/ingest/orm/node.py:94]: What `pb` stands for ?

[the remark this answers is not in the document]
Comment (lferran) [reply nucliadb/nucliadb/ingest/orm/node.py]: protobuffer

**Review by lferran — APPROVED** [review]

**Pull request merged**